astropylibrarian/algolia/records.py

```diff
@@ -1,110 +1,314 @@
+    )
+
+    python_package_keywords: Optional[List[str]] = Field(
+        description="List of python package keywords.",
+    )
+
+    task_keywords: Optional[List[str]] = Field(
+        description="List of task keywords."
+    )
+
+    science_keywords: Optional[List[str]] = Field(
+        description="List of science keywords."
+    )
+
+    content_type: ContentType = Field(
+        description="Content type.", default=ContentType.tutorial
+    )
+
+    @validator("content_type")
+    def validate_content_type(cls, v: Optional[str]) -> str:
+        if v is None:
+            return ContentType.tutorial
+        elif v != ContentType.tutorial:
+            raise ValueError("Content type must be `tutorial`.")
+        return v
+
+    @classmethod
+    def from_section(
+        cls,
+        *,
+        tutorial: ReducedTutorial,
+        section: Section,
+        keyworddb: KeywordDb,
+        index_epoch: str,
+    ) -> TutorialRecord:
+        """Create a TutorialRecord from a reduced tutorial HTML page and
+        specific section.
+
+        Parameters
+        ----------
+        tutorial : astropylibrarian.reducers.tutorial.ReducedTutorial
+            The tutorial.
+        section : astropylibrarian.reducers.utils.Section
+            A section of the tutorial, which corresponds 1:1 with an Algolia
+            record.
+        keyworddb : astropylibrarian.keywords.KeywordDb
+            The keyword database to sort keywords in tutorials into
+            categories for the Learn Astropy UI.
+        index_epoch
+            A unique identifier for the indexing run.
+
+        Returns
+        -------
+        TutorialRecord
+            A tutorial record, ready to index in Algolia.
+        """
+        base_url = cls.compute_base_url(tutorial=tutorial)
+        kwargs: Dict[str, Any] = {
+            "objectID": cls.compute_object_id_for_section(section),
+            "index_epoch": index_epoch,
+            "url": section.url,
+            "root_url": base_url,
+            "root_title": tutorial.h1,
+            "root_summary": tutorial.summary,
+            "base_url": base_url,
+            "importance": section.header_level,
+            "content": section.content,
+            "authors": tutorial.authors,
+            "astropy_package_keywords": keyworddb.filter_keywords(
+                tutorial.keywords, "astropy_package"
+            ),
+            "python_package_keywords": keyworddb.filter_keywords(
+                tutorial.keywords, "python_package"
+            ),
+            "task_keywords": keyworddb.filter_keywords(
+                tutorial.keywords, "task"
+            ),
+            "science_keywords": keyworddb.filter_keywords(
+                tutorial.keywords, "science"
+            ),
+        }
+        for i, heading in enumerate(section.headings):
+            kwargs[f"h{i+1}"] = heading
+        if tutorial.images:
+            kwargs["thumbnail_url"] = tutorial.images[0]
```

Comment:
Add a TODO comment here to note that we might want to make this customizable via notebook metadata? Also will this fail if there are no images?